src/main/java/com/serhiiklymchuk/gmailpet/util/mapper/MessageToMessageDtoMapper.java

```diff
@@ -0,0 +1,63 @@
+package com.serhiiklymchuk.gmailpet.util.mapper;
+
+import com.serhiiklymchuk.gmailpet.domain.Message;
+import com.serhiiklymchuk.gmailpet.domain.User;
+import com.serhiiklymchuk.gmailpet.dto.MessageDto;
+import com.serhiiklymchuk.gmailpet.repository.UserRepository;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class MessageToMessageDtoMapper {
+
+    private final UserRepository userRepository;
+
+    public MessageToMessageDtoMapper(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    private MessageDto map(Message message, User receiverUser, User senderUser) {
+
+        return MessageDto.builder()
+                .receiverUsername(receiverUser.getUsername())
+                .senderUsername(senderUser.getUsername())
+                .subject(message.getSubject())
+                .content(message.getContent())
+                .reviewed(message.isReviewed())
+                .date(message.getDate())
+                .build();
+    }
+
+    public List<MessageDto> map(List<Message> messages, User user) {
```

Comment:
The idea should be to query all users in two queries here and the use inmemory collection to perform mapping.
Current approach will still perform a lot of queries